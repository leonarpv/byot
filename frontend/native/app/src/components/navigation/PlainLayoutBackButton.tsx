import * as React from 'react';
import {Animated, StyleSheet, TouchableOpacity} from 'react-native';
import {KeyboardArrowLeft} from '../icons/material/KeyboardArrowLeft';
import {Colors} from 'react-native-ui-lib';

interface Props {
  onPress: () => void;
  visible?: boolean;
}

interface State {}

const makeStyles = (props: Props, state: State) =>
  StyleSheet.create({
    root: {
      position: 'absolute',
      left: 0,
      top: 0,
    },
    icon: {
      color: Colors.primary,
      fontSize: 48,
    },
  });

export const PlainLayoutBackButton: React.FC<Props> = (props: Props) => {
  const styles = makeStyles(props, {});
  const fade = React.useRef(new Animated.Value(props.visible ? 1 : 0)).current;
  const fadeIn = () =>
    Animated.timing(fade, {
      toValue: 1,
      duration: 500,
      useNativeDriver: true,
    }).start();

  const fadeOut = () =>
    Animated.timing(fade, {
      toValue: 0,
      duration: 200,
      useNativeDriver: true,
    }).start();
  React.useEffect(() => {
    if (props.visible) {
      fadeIn();
    } else {
      fadeOut();
    }
  }, [props.visible]);
  return (
    <Animated.View style={[styles.root, {opacity: fade}]}>
      <TouchableOpacity onPress={props.onPress}>
        <KeyboardArrowLeft style={styles.icon} />
      </TouchableOpacity>
    </Animated.View>
  );
};
