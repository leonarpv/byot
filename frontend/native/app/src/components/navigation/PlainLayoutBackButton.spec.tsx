import * as React from 'react';
import {render} from '@testing-library/react-native';
import {PlainLayoutBackButton} from './PlainLayoutBackButton';

describe('<PlainLayoutBackButton/>', () => {
  it('should render', () => {
    const {asJSON} = render(<PlainLayoutBackButton onPress={jest.fn()} />);
    expect(asJSON()).toMatchSnapshot();
  });
});
