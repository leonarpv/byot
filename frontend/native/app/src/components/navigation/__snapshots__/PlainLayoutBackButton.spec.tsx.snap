// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlainLayoutBackButton/> should render 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <TouchableOpacity>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#FAFAF1",
              "fontSize": 48,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        îŒ”
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
